.azure-pipelines/scripts/checkout_submodules.sh

```diff
@@ -1,8 +1,11 @@
 #!/bin/bash
 
+set -ex
+
 mkdir -p ~/.ssh > /dev/null
 chmod 700 ~/.ssh
 ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+find .git -type f -name '*.lock' -exec rm -v {} +
```

Comment:
Would it also be a good idea to kill any leftover git processes on the machine?

Reply:
No, in my experience git processes never "hang" or similar, so there is no need to check if any are still running. But if something goes bad then they may leave lock files behind.